website/docs/d/google_impersonated_credential.html.markdown

```diff
@@ -0,0 +1,96 @@
+---
+layout: "google"
+page_title: "Google: google_impersonated_credential"
+sidebar_current: "docs-google-impersonated-credential"
+description: |-
+  Produces access_token for impersonated service accounts
+---
+
+# google\_impersonated\_credential
+
+This data source provides a google `oauth2` `access_token` for a different service account than the one initially running the script.  You can 
+then use this new token to access resources the original caller would not have permissions on otherwise.
+
+For more information see
+[the official documentation](https://cloud.google.com/iam/docs/creating-short-lived-service-account-credentials) as well as [iamcredentials.generateAccessToken()](https://cloud.google.com/iam/credentials/reference/rest/v1/projects.serviceAccounts/generateAccessToken)
+
+## Example Usage
+
+To allow `service_A` to impersonate `service_B`, grant the [Service Account Token Creator](https://cloud.google.com/iam/docs/service-accounts#the_service_account_token_creator_role) on B to A. 
+
+In the IAM policy below, `service_A` is given the Token Creator role impersonate `service_B`
+
+```hcl
+$ cat service_policy.json 
+{ 
+"bindings": [
+    {
+      "members": [
+        "serviceAccount:service_A@projectA.iam.gserviceaccount.com"
+      ],
+      "role": "roles/iam.serviceAccountTokenCreator",    
+    }
+  ]
+}
+
+$ gcloud iam service-accounts set-iam-policy  service_B@projectB.iam.gserviceaccount.com  service_policy.json
```

Comment:
Can you use terraform to set up the permissions rather than `gcloud` in the example? It seems strange to tell users to manage permissions with a different tool when terraform supports this.

Reply:
Alternatively you could just instruct them to "set serviceAccountTokenCreator permission on the first account" without the step by step.